FirebaseCombineSwift/Tests/Unit/Firestore/AddDocumentTests.swift

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import FirebaseCombineSwift
+import FirebaseFirestore
+import Combine
+import XCTest
+
+class AddDocumentTests: XCTestCase {
+    
+    class MockCollectionReference: CollectionReference {
+        
+        var mockAddDocument: () throws -> Void = {
+            fatalError("You need to implement \(#function) in your mock.")
+        }
+        
+        var verifyData: ((_ data: [String : Any]) throws -> Void)?
+        
+        override func addDocument(data: [String : Any], completion: ((Error?) -> Void)? = nil) -> DocumentReference {
+            do {
+                try verifyData?(data)
+                try mockAddDocument()
+                completion?(nil)
+            } catch {
+                completion?(error)
+            }
+            return document()
+        }
+        
+        override init() {
+        }
```

Comment:
:rotating_light: Error here.

Reply:
Trying to come up with a solution for this (see #7632, which is a WIP as well...).